Update the technical glossary (#44)
massalabs/massa-docs: technical-doc/glossary.rst

Where is the money?
===================

- **wallet**: A set of keypairs (with their associated address).
- **address**: Hash of a public key preceded by an indentifier and a version bit. Tokens, rolls, bytecode and datastore are associated with an address. An address is associated with a thread for sharding purposes.  
- **smart contract address**: A specific kind of address that was generated by bytecode execution, that is not associated with a keypair. Hence it is immutable. Note: a deletion mechanism can be implemented in the associated byte code.
- **final balance**: Balance at the latest final blocks.
- **candidate balance**: Balance at latest blocks.
- **deferred credits**: Coins to be reimbursed upon a given slot. Produced by a roll sell or a roll slash.
- **ledger vs balance**: The ledger is the long term memory of the blockchain. For every address it may contain a balance, bytecode and a datastore (key / value container).
- **staking address**: Address that has at least one roll.

Graph stuff
===========

> - **compatibility graph**: Graph of all compatible blocks (see https://arxiv.org/pdf/1803.09029.pdf at III. ARCHITECTURE, D. Consensus Rule, 1. Compatibility Graph). Theoretical construction. Never computed by the Massa node.

Reviewer: ```suggestion
- **compatibility graph**: A graph of all compatible blocks (see https://arxiv.org/pdf/1803.09029.pdf at III. ARCHITECTURE, D. Consensus Rule, 1. Compatibility Graph). Theoretical construction. Never computed by the Massa node.
```

> - **head incompatibility graph**: Graph of incompatible active blocks (see https://arxiv.org/pdf/1803.09029.pdf). Computed by the Massa node. Simpler to compute than the compatibility graph, practical application of it.

Reviewer: ```suggestion
- **head incompatibility graph**: A graph of incompatible active blocks (see https://arxiv.org/pdf/1803.09029.pdf). Computed by the Massa node. Simpler to compute than the compatibility graph, practical application of it.
```

> - **fitness**: For a block: 1 + number of endorsement included in the block. For a clique: sum of its blocks fitness.

Reviewer: ```suggestion
- **fitness**: For a block: 1 + number of endorsements included in the block. For a clique: sum of its blocks' fitness.
```
